### PR TITLE
use RaftEngine in debug service

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Use RaftEngine in debug service.

### Check List 

Tests 

- Unit test

Side effects

### Release note 
* No release note

### CODE REVIEW

File: cmd/Cargo.toml [hunk 1]
```diff
@@ -85,6 +85,7 @@ tikv_util = { path = "../components/tikv_util" }
 toml = "0.4"
 txn_types = { path = "../components/txn_types" }
 vlog = "0.1.4"
+raft-engine = { git = "https://github.com/tikv/raft-engine" }
```
Comment (Little-Wallace) [inline cmd/Cargo.toml:88]: It seems that `raft-engine` has not been used still?

File: src/server/service/debug.rs [hunk 2]
```diff
@@ -366,8 +366,8 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static> debugpb::Debug for Service<T> {
             resp.set_prometheus(metrics::dump());
             if req.get_all() {
                 let engines = debugger.get_engine();
-                resp.set_rocksdb_kv(box_try!(engines.kv.dump_stats()));
-                resp.set_rocksdb_raft(box_try!(engines.raft.dump_stats()));
+                resp.set_rocksdb_kv(box_try!(MiscExt::dump_stats(&engines.kv)));
+                resp.set_rocksdb_raft(box_try!(RaftEngine::dump_stats(&engines.kv)));
```
Comment (Little-Wallace) [inline src/server/service/debug.rs:370]: It should be raft than kv?

File: components/raft_log_engine/src/engine.rs [hunk 3]
```diff
@@ -165,4 +165,8 @@ impl RaftEngine for RaftLogEngine {
     fn stop(&self) {
         self.0.stop();
     }
+
+    fn dump_stats(&self) -> Result<String> {
+        Ok("".to_owned())
```
Comment (NingLin-P) [inline components/raft_log_engine/src/engine.rs:170]: Should add a TODO comment if plan to support it in the future.

**Review by NingLin-P — APPROVED** [review]
Rest LGTM

File: Cargo.toml [hunk 4]
```diff
@@ -81,7 +81,7 @@ engine_rocks = { path = "components/engine_rocks" }
 engine_traits = { path = "components/engine_traits" }
 error_code = { path = "components/error_code" }
 fail = "0.3"
-raft-engine = { git = "https://github.com/tikv/raft-engine", default-features = false }
+raft-engine = { git = "https://github.com/tikv/raft-engine"}
```
Comment (Little-Wallace) [inline Cargo.toml:84]: I think tikv could depend `component/raft_log_engine` and not need to depend `raft-engine` directly.

File: components/raft_log_engine/src/engine.rs (hunk 3 above)
Comment (hicqu) [reply components/raft_log_engine/src/engine.rs:170]: I will leave a comment here.

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**